app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt

```diff
@@ -313,6 +352,16 @@ class UrlInputFragment :
             view = browserToolbar
         )
 
+        topSitesFeature.set(
+            feature = TopSitesFeature(
+                view = DefaultTopSitesView(requireComponents.appStore),
+                storage = requireComponents.topSitesStorage,
+                config = ::getTopSitesConfig
+            ),
+            owner = this,
+            view = view
```

Comment:
This seems a bit hacky. If I understand it correctly this feature will load top sites and then pass them to the view, and we have a "fake" view implementation that will dispatch it on `AppStore`?

I guess ideally we wouldn't have to go through this `DefaultTopSitesStorage`manually  at all. Dispatch an action to update the state and a middleware updates the storage. But maybe to big of a change right now.